app/forms/waste_exemptions_engine/registration_number_form.rb

```diff
@@ -28,10 +28,10 @@ def submit(params)
     private
 
     def process_company_no(company_no)
-      number = company_no.to_s
+      number = company_no.to_s.strip
```

Comment:
Is this duplicating work which the CanStripWhitespace concern should be doing? I think all attributes should be stripped before validation.

Reply:
Yes, you are right. Wonder why it's not picking it up. Will investigate and come back with edits.

Reply:
Ahhhh! Now know why its not picking it up. `CanStripWhitespace` is a concern added to the `BaseForm`. Each form will sort out its own attributes (in this case it's just `company_no`) and then pass them to the base `submit()` method. In there is passes them to the concern's `strip_whitespace()` method.

All good so far, we're ensuring the values we save to the database are stripped of whitespace. The issue I've realised is we never update the properties of the form afterwards, which were originally used to create the attributes. And it's the form and its properties all validators look at.

This is why the value we save is stripped, but the value we validate is not.

As our intent is to strip whitespace before both saving and validating, I think this will require a bit more thought before we do anything as to what the solution will be, because all existing form objects are affected by this in both services (WCR & WEX).